core/src/main/scala/akka/kafka/Metadata.scala

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (C) 2014 - 2016 Softwaremill <http://softwaremill.com>
+ * Copyright (C) 2016 - 2018 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.kafka
+
+import java.util.Optional
+
+import akka.actor.NoSerializationVerificationNeeded
+import org.apache.kafka.clients.consumer.{OffsetAndMetadata, OffsetAndTimestamp}
+import org.apache.kafka.common.{PartitionInfo, TopicPartition}
+
+import scala.collection.JavaConverters._
+import scala.util.Try
+
+/**
+ * Messages for Kafka metadata fetching via [[KafkaConsumerActor]].
+ *
+ * NOTE: Processing of these requests blocks the actor loop. The KafkaConsumerActor is configured to run on its
+ * own dispatcher, so just as the other remote calls to Kafka, the blocking happens within a designated thread pool.
+ * However, calling these during consuming might affect performance and even cause timeouts in extreme cases.
+ */
+object Metadata {
+
+  sealed trait Request
+  sealed trait Response
+
+  /**
+   * [[org.apache.kafka.clients.consumer.KafkaConsumer#listTopics()]]
+   */
+  case object ListTopics extends Request with NoSerializationVerificationNeeded
```

Comment:
This will require a getter for the Java API: `getInstance()`

Reply:
I'll call it createListTopics in line with the case class constructing things.